### PR TITLE
Search: Update upsell page for pricing change

### LINKED ISSUE #25915: Search Plugin: Add plans view page
The page is to replace the old [upsell-page.jsx](https://github.com/Automattic/jetpack/blob/0104f558800dcee071842fb5f3cb78fa21449368/projects/packages/search/src/dashboard/components/pages/upsell-page.jsx)

image

Reference: `p1HpG7-hvT-p2`

### PR DESCRIPTION
**Pull request by dognose24**
Fixes #25915 

#### Changes proposed in this Pull Request:

* Expose the `offset` option of the `IconTooltip` component to the Popover component and fix the relative position
* Modify `PricingTable` component styles adapting to the design and refactor the type of `tooltipInfo` to ReactNode for HTML syntax
* Introduce PricingTable to complete the upsell page for the new pricing change
-------------------------------------- updated --------------------------------------
* Introduce the URL parameter `new_pricing_202208=1` to determine when the new pricing table shows.
#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

p1HpG7-hEH-p2

#### Does this pull request change what data or activity we track or use?

N/A

#### Testing instructions:

* Go to a site with `Jetpack` connected but did not purchase `Jetpack Search`
* Navigate to the url `/wp-admin/admin.php?page=jetpack-search`
* Ensure there is the existing upsell page with a pricing card
* Navigate to the url `/wp-admin/admin.php?page=jetpack-search&new_pricing_202208=1`
* Ensure there is a brand new upsell page

| Before | After |
| --- | ----------- |
| 截圖 2022-09-24 上午5 05 00 | 截圖 2022-09-24 上午5 22 08 |

* Ensure the new upsell page conforms to the design
截圖 2022-09-24 上午5 08 32

### CODE REVIEW

File: projects/packages/search/src/dashboard/components/pages/upsell-page.jsx [hunk 1]
```diff
@@ -51,10 +55,173 @@ export default function UpsellPage( { isLoading = false } ) {
...
+				tooltipInfo: (
+					<>
+						Quick and accurate spelling correction for when your site visitors mistype their search.
+					</>
+				),
+			},
+		],
+	};
+
+	const pricingTable = (
```
Comment (jsnmoon) [inline projects/packages/search/src/dashboard/components/pages/upsell-page.jsx]: We can spin this off into a functional component (e.g. `SearchPricingTable`) in this file. Let's tackle this in a follow-up PR.

File: projects/js-packages/components/components/product-price/index.tsx [hunk 2]
```diff
@@ -46,7 +47,8 @@ const ProductPrice: React.FC< ProductPriceProps > = ( {
 				) }
 				{ promoLabel && <Text className={ styles[ 'promo-label' ] }>{ promoLabel }</Text> }
 			</div>
-			{ <Text className={ styles.leyend }>{ leyend }</Text> }
+			{ ! children && <Text className={ styles.leyend }>{ leyend }</Text> }
+			{ children }
```
Comment (jsnmoon) [inline projects/js-packages/components/components/product-price/index.tsx]: Something to tackle in a follow-up PR:

```suggestion
			{ children ? children : <Text className={ styles.leyend }>{ leyend }</Text> }
```

File: projects/packages/search/src/dashboard/components/pages/upsell-page.jsx [hunk 3]
```diff
@@ -69,24 +236,9 @@ export default function UpsellPage( { isLoading = false } ) {
...
-										priceAfter={ priceAfter }
-										priceBefore={ priceBefore }
-										currencyCode={ priceCurrencyCode }
-										title={ __( 'Jetpack Search', 'jetpack-search-pkg' ) }
-									/>
+							<Container horizontalSpacing={ 8 }>
+								<Col lg={ 12 } md={ 12 } sm={ 12 }>
+									<ThemeProvider>{ pricingTable }</ThemeProvider>
 								</Col>
 							</Container>
```
Comment (jsnmoon) [inline projects/packages/search/src/dashboard/components/pages/upsell-page.jsx]: We don't want to remove the existing upsell interface until we launch the new pricing tiers. Instead, we'll want to conditionally render either the old or the new upsell screen using a feature gate.

To do this, I recommend using our dashboard's [built-in feature flag capability](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/search/src/dashboard/class-initial-state.php#L85) which will look for a comma separated value stored in the `features` query string. For example, navigating to:

`/wp-admin/admin.php?page=jetpack-search&features=pricing,pricing2`

will store a value of `['pricing', 'pricing2']` in `window.JETPACK_SEARCH_DASHBOARD_INITIAL_STATE.features`.

I imagine our implementation might look something like this:
```
{ isNewPricingEnabled ? <NewUpsellContainer /> : <OldUpsellContainer /> }
```

---

@kangzj [recently shipped code for feature gating the new pricing logic](https://github.com/Automattic/jetpack/pull/26338), so he might have a better alternative than what I proposed here.

**Review by jsnmoon — CHANGES_REQUESTED** [review]
Unfortunately, this PR removes the existing upsell interface before the new pricing tier goes live. I commented inline on how to ship this code while preserving our current onboarding flow.

File: projects/js-packages/components/components/product-price/index.tsx (hunk 2 above)
Comment (oskosk) [reply projects/js-packages/components/components/product-price/index.tsx]: Feedback addressed in 4b8c663

**Review by jsnmoon — APPROVED** [review]
Works as described. 

I noticed that the links within the tooltips aren't clickable since the tooltips despawn when we stop mousing over the icon. Let's address this in a follow-up PR 🙂

File: projects/packages/search/src/dashboard/components/pages/upsell-page.jsx (hunk 3 above)
Comment (oskosk) [reply projects/packages/search/src/dashboard/components/pages/upsell-page.jsx]: Addressed in [7017e1b](https://github.com/Automattic/jetpack/pull/26408/commits/7017e1ba4d1f5a47716bc6442ae78614c2052dfa)

**Review by oskosk — APPROVED** [review]
Approving again to see if it stops being stuck

**Comment by jsnmoon** [comment]
🚢 r253223-wpcom

**Comment by jsnmoon** [comment]
> I noticed that the links within the tooltips aren't clickable since the tooltips despawn when we stop mousing over the icon. Let's address this in a follow-up PR 🙂

Fix for this tooltip bug is proposed [here](https://github.com/Automattic/jetpack/pull/26457).

**Pull request merged**